Sources/OpenAPIKit/Schema Object/JSONSchema+Combining.swift

```diff
@@ -559,12 +559,16 @@ extension JSONSchema.ObjectContext {
     }
 }
 
-internal func combine(properties left: [String: JSONSchema], with right: [String: JSONSchema], resolvingIn components: OpenAPI.Components) throws -> [String: JSONSchema] {
-    var combined = left
-    try combined.merge(right) { (left, right) throws -> JSONSchema in
-        var resolver = FragmentCombiner(components: components)
-        try resolver.combine([left, right])
-        return try resolver.dereferencedSchema().jsonSchema
+internal func combine(properties left: OrderedDictionary<String, JSONSchema>, with right: OrderedDictionary<String, JSONSchema>, resolvingIn components: OpenAPI.Components) throws -> OrderedDictionary<String, JSONSchema> {
+    var combined = right
```

Comment:
I'm not sure about this part. Based on one of the unit test `test_allOfExample`, it produces the correct result.

Input schema:

```yaml
Dog:
    allOf:
        - $ref: '#/components/schemas/Pet'
        - type: object
          properties:
                bark:
                    type: boolean
                breed:
                    type: string
                    enum: [Dingo, Husky, Retriever, Shepherd]
```

Output:

```swift
let dogSchema = JSONSchema.object(
    discriminator: .init(propertyName: "pet_type"),
    properties: [
        "pet_type": .string,
        "bark": .boolean(required: false),
        "breed": .string(required: false, allowedValues: "Dingo", "Husky", "Retriever", "Shepherd")
    ]
)
```

But I don't understand why. The framework puts `schemas/Pet` as a right item. I don't know if that's coincidental or by design.

Reply:
> it produces the correct result (or at least the result I would expect).

I really hope that at least it produces a semantically identical result even if the order of things is different (I don't think I've written any code in this file that would produce _truly_ different results by combining `a` with `b` vs. `b` with `a`.

As for the order being what it is, it's coincidental but not without explanation. There's a massive `switch` statement in the `combine` function on line 115 of this file that is only even as small as it is because I have only represented half of the possible cases. Specifically, I've only represented the cases where the `left` thing is "less specialized" than the `right` thing. Incidentally, this choice carries through to another part of the file where the function you are modifying here is called from. It could be swapped, but the intuition you had about the order in this case may or may not always hold anyway no matter which way you swap them because one of the two will always be the less specialized owing to the way I wrote the massive switch statement.

In the example you are referencing, I don't think either of the two will be considered less specialized than the other, so perhaps the intuitive thing to do would be swap left & right earlier on, but again I don't think you will be able to count on that intuition always holding when you do have more & less specialized schemas being combined.